src/components/StudioTopbar/StudioTopbar.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { useActiveUser } from '@/hooks'
+import routes from '@/config/routes'
+import { Text, Icon, Button } from '@/shared/components'
+import {
+  StyledTopbarBase,
+  StudioContainer,
+  DrawerButton,
+  MemberInfoContainer,
+  MemberTitleText,
+  MemberTextContainer,
+  ChannelInfoContainer,
+  StyledAvatar,
+  TextContainer,
+  DrawerContainer,
+  NewChannel,
+  NewChannelIconContainer,
+  StyledLink,
+} from './StudioTopbar.style'
+
+type Channel = {
+  name: string
+  avatar: string
+}
+type ChannelInfoProps = {
+  active?: boolean
+  member: string
+  channel: Channel
+  onClick?: React.MouseEventHandler<HTMLDivElement>
+}
+
+type MemberInfoProps = {
+  memberName: string
+  memberAvatar: string
+}
+
+type NavDrawerProps = {
+  active?: boolean
+  memberName: string
+  memberAvatar: string
+  channels: Channel[]
+  currentChannel: Channel
+  onCurrentChannelChange: (channel: Channel) => void
+  onLogoutClick: () => void
+  handleClose: () => void
+}
+
+const member = {
+  name: 'Mikael Cowan',
+  avatar: 'https://picsum.photos/300/300',
+  channels: [
+    { name: 'Wild Crypto Fan16', avatar: 'https://picsum.photos/201/300' },
+    { name: 'Mild Crypto Skeptic', avatar: 'https://picsum.photos/202/300' },
+    { name: 'Average Cryptocurrency Enjoyer', avatar: 'https://picsum.photos/203/300' },
+  ],
+}
+
+const StudioTopbar: React.FC = () => {
+  const { activeUser, setActiveChannel } = useActiveUser()
+  // TODO Add member fetching
+  const [currentChannel, setCurrentChannel] = useState(member.channels[0])
+
+  const [isDrawerActive, setDrawerActive] = useState(false)
+
+  const drawerRef = useRef<HTMLDivElement | null>(null)
+  const channelInfoRef = useRef<HTMLDivElement | null>(null)
+  const drawerButtonRef = useRef<HTMLButtonElement | null>(null)
+
+  const handleCurrentChannelChange: (channel: Channel) => void = (channel) => {
+    setCurrentChannel(channel)
+    setDrawerActive(false)
+  }
+
+  const handleLogout = () => {
+    // TODO add logic for Logout
+    window.alert("You've been logged out")
+  }
+
+  const handleDrawerToggle: (e: React.MouseEvent<HTMLElement>) => void = (e) => {
+    e.stopPropagation()
+    setDrawerActive(!isDrawerActive)
+  }
+
+  useEffect(() => {
+    const handleClickOutside = (event: Event) => {
+      if (
+        channelInfoRef.current?.contains(event.target as Node) ||
+        drawerButtonRef.current?.contains(event.target as Node)
+      ) {
+        return
+      }
+      if (drawerRef.current && !drawerRef.current.contains(event.target as Node)) {
+        setDrawerActive(false)
+      }
+    }
+    document.addEventListener('click', handleClickOutside, true)
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true)
+    }
+  }, [])
```

Comment:
I think we potentially could improve this. This will add an event listener on mount and react on every page click. So regardless of whether the overlay is open or not, we will constantly call `setDrawerActive(false)` on every page click. What we could is add event listener only once the overlay is opened. Then we could also simplify the logic here I think. Not sure this will work 1:1, but consider this:
```suggestion
  useEffect(() => {
    if (!isDrawerActive) {
      return
    }

    const handleClickOutside = (event: Event) => {
      if (drawerRef.current && !drawerRef.current.contains(event.target as Node)) {
        // stop propagation so drawer doesn't get triggered again on button click
        event.stopPropagation()
        setDrawerActive(false)
      }
    }
    document.addEventListener('click', handleClickOutside, true)
    return () => {
      document.removeEventListener('click', handleClickOutside, true)
    }
  }, [isDrawerActive])
```

I think this also allows us to get rid of `channelInfoRef` and `drawerButtonRef`. Let me know if that works